### ISSUE TITLE
Running CityScapes twice gives different results

### ISSUE THREAD

**Issue by penguinmenac3** [bonlime/keras-deeplab-v3-plus#86]
When I run a script (very similar to the suggestion in the README.md), I get different outputs every time I execute it.

I am not talking about slightly different outputs, but the class mapping being completely messed up.

![Figure_1](https://user-images.githubusercontent.com/4391367/56362579-813cd580-61ea-11e9-80e5-c7a00ad73de3.png)
![Figure_2](https://user-images.githubusercontent.com/4391367/56362581-81d56c00-61ea-11e9-83e2-facadc525955.png)

At first I thought this was just matplotlib generating different mappings each run, however, after applying the colormapping from deeplabs code, I found the problem persistet.

## For Reproducability

I am running `tensorflow-gpu==2.0.0-alpha0 `
Here is the script:

```python
from matplotlib import pyplot as plt
import cv2 # used for resize. if you dont have it, use anything else
import numpy as np
import tensorflow as tf
from model import Deeplabv3

# Load stuff
model = Deeplabv3(weights='cityscapes')
image = plt.imread("imgs/image1.jpg")

# Prepare the image
w, h, _ = image.shape
ratio = 512. / np.max([w,h])
resized = cv2.resize(image,(int(ratio*h),int(ratio*w)))
resized = resized / 127.5 - 1.
pad_x = int(512 - resized.shape[0])
resized2 = np.pad(resized,((0,pad_x),(0,0),(0,0)),mode='constant')

# Predict
res = model.predict(np.expand_dims(resized2,0))
labels = np.argmax(res.squeeze(),-1)
labels = labels[:-pad_x]

# Create colormap as defined in deeplab (https://github.com/tensorflow/models/blob/17e923da9e8caba5dfbd58846ce75962206ffa64/research/deeplab/utils/get_dataset_colormap.py#L207)
colormap = np.zeros((256, 3), dtype=np.uint8)
colormap[0] = [128, 64, 128]
colormap[1] = [244, 35, 232]
colormap[2] = [70, 70, 70]
colormap[3] = [102, 102, 156]
colormap[4] = [190, 153, 153]
colormap[5] = [153, 153, 153]
colormap[6] = [250, 170, 30]
colormap[7] = [220, 220, 0]
colormap[8] = [107, 142, 35]
colormap[9] = [152, 251, 152]
colormap[10] = [70, 130, 180]
colormap[11] = [220, 20, 60]
colormap[12] = [255, 0, 0]
colormap[13] = [0, 0, 142]
colormap[14] = [0, 0, 70]
colormap[15] = [0, 60, 100]
colormap[16] = [0, 80, 100]
colormap[17] = [0, 0, 230]
colormap[18] = [119, 11, 32]

# Plot the image, and the labels (colorized using the deeplab colormap)
plt.clf()
plt.subplot(1,2,1)
plt.imshow(image)
plt.subplot(1,2,2)
plt.imshow(colormap[labels[:,:]])
plt.show()
```

**Comment by penguinmenac3** [comment]
Weakly related to #83 and #80.

**Comment by penguinmenac3** [comment]
More or less solved it, it seems as if the weights are interpreted in a weird way when using the wrong number of output classes (default is 21 if you do not set it).

```
model = Deeplabv3(weights=CITYSCAPES, classes=19)
```
It seems it is not possible to load pretrained weights into a model with a different number of outputs as the pretrained model.
The cause is model.py:432-437 since a layer gets a different name here, resulting in reinitializing it with random weights, thus the above reported results.

@bonlime Maybe it is possible to expose an argument to enable/disable renaming of the last layer to custom or not. Some people might want to reinitialize the layer even though they are using by accident 19/21 classes. I would suggest a parameter `custom_logits_semantic=False`.

Feel free to close the issue by either adding such a parameter for explicitly specifying the behaviour of model.py:432-437 or declining the feature request.

**Comment by bonlime** [comment]
Hi. The main reason behind renaming the layer to custom is as follows: saved weights have certain dimensions and if you specify the number of classes different from the saved model and try to load the weights without renaming the last layer, it would cause an error. 

You're right that if you're trying to retrain this model to a certain task with 21 or 19 classes, the weights are going to be pretrained rather that initialized randomly. From my point of view, this case is very rare and doesn't justify adding a new parameter to the model. 
To summarize: renaming the last layer to custom is not something you should have control of but rather a protection from errors while loading the weights. 

And in addition to that. documentation for the `classes` parameter clearly states that "If number of classes not aligned with the weights used, last layer is initialized randomly" so it shouldn't be a real problem.
